src/vcf/validator.cpp

```diff
@@ -135,12 +142,19 @@ namespace ebi
                            ValidationLevel validationLevel,
                            std::vector<std::unique_ptr<ebi::vcf::ReportWriter>> &outputs)
     {
+        boost::iostreams::filtering_istream uncompressed_input;
+        boost::iostreams::filtering_istream input_stream(input);
```

Comment:
filtering_stream::push doesn't make copy of it. filtering_stream is not copy-constructible. So if we declare `input_stream` in function`create_uncompressed_stream` it will be destroyed with the function return and will cause segmentation fault.